docker/Dockerfile

```diff
@@ -33,24 +31,29 @@ ENV CONDA_RUN="conda run --no-capture-output --prefix ${CONDA_PREFIX}"
 ENV PYTHON_VERSION="3.10"
 ENV CONTAINER_PUDL_IN=${CONTAINER_HOME}/pudl_in
 ENV CONTAINER_PUDL_OUT=${CONTAINER_HOME}/pudl_out
+ENV DAGSTER_HOME=${CONTAINER_HOME}/dagster_home
 
 # Create a conda environment based on the specification in the repo
+COPY test/test-environment.yml test/test-environment.yml
+RUN mamba create --copy --prefix ${CONDA_PREFIX} --yes python=${PYTHON_VERSION} && \
+    # Then we can use mamba env update, which can parse the environment.yml file:
+    mamba env update --prefix ${CONDA_PREFIX} --file test/test-environment.yml && \
+    conda clean -afy
+
+# Copy the cloned pudl repository into the user's home directory
+COPY --chown=catalyst:catalyst . ${CONTAINER_HOME}
+
```

Comment:
Why did you move this? I assume it enables more image layer caching?

Reply:
Yep! Basically, since `.` changes all the time, pulling in `.` is going to invalidate the cache for everything after it. But we can copy in the requirements files first, and those don't change as frequently. This lets us skip the ~2m mamba build step every time I try to rebuild the image.